app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/NightscoutUploader.java

```diff
@@ -110,157 +133,73 @@ private boolean doRESTUpload(SharedPreferences prefs, List<BgReading> glucoseDat
-                        Log.i(TAG, "MBG JSON: " + jsonString);
-
-                        try {
-                            StringEntity se = new StringEntity(jsonString);
-                            post.setEntity(se);
-                            post.setHeader("Accept", "application/json");
-                            post.setHeader("Content-type", "application/json");
-
-                            ResponseHandler responseHandler = new BasicResponseHandler();
-                            httpclient.execute(post, responseHandler);
-                        } catch (Exception e) {
-                            Log.e(TAG, "Unable to post data", e);
-                        }
+                    if (apiVersion == 1) {
+                        String hashedSecret = Hashing.sha1().hashBytes(secret.getBytes(Charsets.UTF_8)).toString();
+                        doRESTUploadTo(nightscoutService, hashedSecret, glucoseDataSets, meterRecords, calRecords);
+                    } else {
+                        doLegacyRESTUploadTo(nightscoutService, glucoseDataSets);
                     }
+                } catch (Exception e) {
+                    Log.e(TAG, "Unable to do REST API Upload " + e.getMessage());
+                    e.printStackTrace();
+                    return false;
                 }
+            }
+            return true;
+        }
 
-                if (apiVersion >= 1) {
-                    for (Calibration calRecord : calRecords) {
-
-                        JSONObject json = new JSONObject();
-
-                        try {
-                            populateV1APICalibrationEntry(json, calRecord);
-                        } catch (Exception e) {
-                            Log.e(TAG, "Unable to populate entry", e);
-                            continue;
-                        }
+        private void doLegacyRESTUploadTo(NightscoutService nightscoutService, List<BgReading> glucoseDataSets) throws Exception {
+            for (BgReading record : glucoseDataSets) {
+                Response<ResponseBody> r = nightscoutService.upload(populateLegacyAPIEntry(record)).execute();
+                if (!r.isSuccess()) throw new UploaderException(r.message(), r.code());
 
-                        String jsonString = json.toString();
-                        Log.i(TAG, "CAL JSON: " + jsonString);
+            }
+            postDeviceStatus(nightscoutService, null);
+        }
 
-                        try {
-                            StringEntity se = new StringEntity(jsonString);
-                            post.setEntity(se);
-                            post.setHeader("Accept", "application/json");
-                            post.setHeader("Content-type", "application/json");
+        private void doRESTUploadTo(NightscoutService nightscoutService, String secret, List<BgReading> glucoseDataSets, List<Calibration> meterRecords, List<Calibration> calRecords) throws Exception {
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [The method doRESTUploadTo() has an NPath complexity of 16](https://www.codacy.com/app/matthias-granberry/xDrip-Experimental/file/1296698421/issues/source?bid=2735053&fileBranchId=2947762#l178)